.azure-pipelines.yml

```diff
@@ -0,0 +1,64 @@
+name: $(Build.BuildId)
+trigger:
+  # This is where pull requests from "bors r+" are built.
+  - staging
+  # This is where pull requests from "bors try" are built.
+  - trying
+  # Always build master
+  - master
+pr:
+  # Enable building pull requests.
+  - master
+stages:
+  - stage: Testing
+    jobs:
+      - job:
+        variables:
+          rust: 1.36.0
+        strategy:
+          matrix:
+            Windows 32-bit (MSVC):
+              image: vs2017-win2016
+              target: i686-pc-windows-msvc
+            Windows 64-bit (MSVC):
+              image: vs2017-win2016
+              target: x86_64-pc-windows-msvc
+            Windows 32-bit (MinGW):
+              image: vs2017-win2016
+              target: i686-pc-windows-gnu
+            Windows 64-bit (MinGW):
+              image: vs2017-win2016
+              target: x86_64-pc-windows-gnu
+        pool:
+          vmImage: $(image)
+        steps:
+          - checkout: self
+            fetchDepth: 1
+            path: clap
+            displayName: Checkout repository
+          - task: Cache@2
+            inputs:
+              key: cargo | "$(rust)" | $(target) | Cargo.toml
+              path: C:\Rust\.cargo
+            displayName: Caching cargo
+          - task: Cache@2
+            inputs:
+              key: compiled | "$(rust)" | $(target) | Cargo.toml
+              path: target
+            displayName: Caching compiled
+          - script: rustup default $(rust)-$(target)
+            displayName: Install rust
+          - script: cargo test --all --no-default-features
+            displayName: Test without default features
+            env:
+              RUST_BACKTRACE: full
+          - script: cargo test --all --features "yaml unstable"
+            displayName: Test with most features
+            env:
+              RUST_BACKTRACE: full
+          - script: |
+              rmdir /Q /S C:\Rust\.cargo\registry\src
+              rmdir /Q /S target\.rustc_info.json
+              rmdir /Q /S target\debug\examples
```

Comment:
I don't know the proper Windows commands. If we could copy the same instructions I had in travis, the CI time would become even lesser because the cache size would be lower and thus faster to restore.

Reply:
Do we even need the cache at all? I propose to remove it entirely and check how the build time degrades (if it does) without it and then decide whether we want it or not.

Reply:
I checked. Without this cache, build time was 5 mins more. I was experimenting today all day. I tried to cache `.rustup`, but that actually increased the build time by 1 min. So, I removed it.

Reply:
Well, 5 mins is almost nothing, considering that no-cache builds are consistent, e.g build time will stay the same in future. Cache tends to grow as new rustc versions are released (nightly gets released every day AFAIK).

Reply:
This cache is tied into rust version and then Cargo.toml. So, it shouldn't grow too much.

Reply:
Is it? We had been observing the contrary in structopt (cache grows over time uncontrollably).

Reply:
Because that's travis. Here also, it will grow, but I reduced the growth by a lot by keeping it only for the deps.

Reply:
And then you can use `cargo-sweep` to remove old toolchain builds and it should pretty easy to not make it grow.